cl/scrapers/management/commands/clone_from_cl.py

```diff
@@ -0,0 +1,298 @@
+            print(f"OpinionCluster with id: {self.id} already in local env.")
+            return
+        except OpinionCluster.DoesNotExist:
+
+            s = requests.session()
+            s.headers = {
+                "Authorization": f"Token {os.environ.get('CL_API_TOKEN', '')}"
+            }
+            cluster_endpoint = f"https://www.courtlistener.com/api/rest/v3/clusters/{self.id}/"
+            results = s.get(cluster_endpoint).json()
+
+            cluster_datum = results
+            docket_datum = s.get(cluster_datum["docket"]).json()
+            court_datum = s.get(docket_datum["court"]).json()
+            citation_data = cluster_datum["citations"]
+            opinion_data = cluster_datum["sub_opinions"]
+            # delete resource_uri value generated by DRF
+            del court_datum["resource_uri"]
+            del docket_datum["resource_uri"]
+            del cluster_datum["resource_uri"]
+            # delete fields with fk or m2m relations or unneeded fields
+            del docket_datum["clusters"]
+            del docket_datum["original_court_info"]
+            del docket_datum["absolute_url"]
+            del docket_datum["audio_files"]
+            del docket_datum["tags"]
+            del docket_datum["panel"]
+            del cluster_datum["docket"]
+            del cluster_datum["citations"]
+            del cluster_datum["sub_opinions"]
+            del cluster_datum["absolute_url"]
+            del cluster_datum["panel"]
+            del cluster_datum["non_participating_judges"]
+            with transaction.atomic():
+                # Get or create required court
+                try:
+                    ct = Court.objects.get_or_create(**court_datum)
+                except:
+                    ct = Court.objects.filter(pk=court_datum["id"])
+                docket_datum["court"] = ct[0]
+
+                # Create docket
+                Docket.objects.create(**docket_datum)
+
+                for cite_data in citation_data:
+                    cite_data["cluster_id"] = cluster_datum["id"]
+                cluster_datum["docket_id"] = docket_datum["id"]
+
+                # Create opinion cluster
+                OpinionCluster.objects.create(**cluster_datum)
+
+                for cite_data in citation_data:
+                    # Create citations
+                    cite_data["cluster_id"] = cluster_datum["id"]
+                    Citation.objects.create(**cite_data)
+
+                for op in opinion_data:
+                    op_data = s.get(op).json()
+                    # delete fields with fk or m2m relations or unneeded fields
+                    del op_data["opinions_cited"]
+                    del op_data["cluster"]
+                    del op_data["absolute_url"]
+                    del op_data["resource_uri"]
+                    del op_data["author"]
+                    del op_data["joined_by"]
+                    op_data["cluster_id"] = cluster_datum["id"]
```

Comment:
Shouldn't this go above with the similar code?

Reply:
It works this way because an opinion cluster can have many opinions, and we need to delete those keys from each opinion

Reply:
Sure, but I'd still expect to see all the `del` work getting done first, not some here, some there, particularly because you're in a transaction block now. Ideally when you're in the transaction you shouldn't be doing very much beyond saving things to the DB. If this change means you need a for-loop up above and a second one here, that's OK.